### DISCUSSION TITLE
Calculating Solar eclipse visibility on earth

### DISCUSSION THREAD

**Discussion by JayaShakthi97** [cosinekitty/astronomy#119] [Q&A] [answered]
_Hi, You have done a great job here!_
I am searching for a way to calculate the positions(lat, long) that a particular solar/lunar eclipse can be seen. I know there are already developed websites out there. But I'm interested in developing a website by myself with some fancy features.
I am planning on using Python for the backend for all the calculations. I saw your examples for calculating future eclipses and planets' positions. I guess these may be useful in my task.
Could you please give me guidelines to proceed with my idea?
Thanks in advance.

**Comment by cosinekitty** [comment, marked as answer]
Hi @JayaShakthi97 ,

Thank you for your kind words. Have you taken a look at the documentation for [eclipse functions](https://github.com/cosinekitty/astronomy/tree/master/source/python#eclipses-and-transits) in the Python version of Astronomy Engine?

In general, lunar eclipses are going to be simpler because anyone who can see the moon during the eclipse can observe the eclipse. Solar eclipses are much more complicated because you have to calculate the fast-moving and ever-changing shape of the shadow on the Earth's surface.

The solar eclipse functions I created come in "global" and "local" versions. The global version finds when and where on the globe the peak part of the eclipse occurs. The local version is given a desired observing location and finds information about solar eclipses as visible at that one location. Astronomy Engine currently does not provide a general map-building functionality like some websites have. That would be far more complicated!

I hope this helps you get some ideas of how to get started. Please let me know if I can help with specific use cases.

**Reply by JayaShakthi97** [reply]
Hi @cosinekitty,

Appreciate the quick response.

Yea, I went through the documentation and seems like I can start with your functionalities. I'd start implementations and reach you for any help.

Thanks again! Stay safe!
